packages/vanilla-router/src/types.ts

```
@@ -0,0 +1,13 @@
+// Vanilla component types
+// Components receive router as parameter and return a render function with no parameters
+export type VanillaComponent = (
+  router: import('@tanstack/router-core').AnyRouter,
```

ESLint

[error] `import()` type annotations are forbidden.

(@typescript-eslint/consistent-type-imports)